integration-tests-2.2/src/test/java/org/openmrs/module/fhir2/provider/r3/MedicationRequestFhirResourceProvider_2_2IntegrationTest.java

```diff
@@ -0,0 +1,68 @@
+package org.openmrs.module.fhir2.provider.r3;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.openmrs.module.fhir2.FhirConstants.OPENMRS_FHIR_EXT_MEDICATION_REQUEST_FULFILLER_STATUS;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import org.hl7.fhir.dstu3.model.Extension;
+import org.hl7.fhir.dstu3.model.MedicationRequest;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.module.fhir2.BaseFhirIntegrationTest;
+import org.openmrs.module.fhir2.providers.r3.BaseFhirR3IntegrationTest;
+import org.openmrs.module.fhir2.providers.r4.MedicationRequestFhirResourceProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+public class MedicationRequestFhirResourceProvider_2_2IntegrationTest extends BaseFhirR3IntegrationTest<MedicationRequestFhirResourceProvider, MedicationRequest> {
+	
+	private static final String MEDICATION_REQUEST_UUID = "6d0ae116-707a-4629-9850-f15206e63ab0";
+	
+	private static final String MEDICATION_REQUEST_DATA_XML = "org/openmrs/module/fhir2/api/dao/impl/FhirMedicationRequest_2_2_initial_data.xml";
+	
+	@Getter(AccessLevel.PUBLIC)
+	@Autowired
+	private MedicationRequestFhirResourceProvider resourceProvider;
+	
+	@Before
+	@Override
+	public void setup() throws Exception {
+		super.setup();
+		executeDataSet(MEDICATION_REQUEST_DATA_XML);
+	}
+	
+	@Test
+	public void shouldReturnExistingMedicationRequestAsJson() throws Exception {
```

Comment:
So, this R3 integration *does* fail, but it fails regardless of whether the 2_2 extension has been added.

The error:  "ca.uhn.fhir.context.ConfigurationException: This context is for FHIR version "DSTU3" but the class "org.hl7.fhir.r4.model.MedicationRequest" is for version "R4""

I'm assuming this because the base translator instantiates an R4 object:

https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/MedicationRequestTranslatorImpl.java#L85

But this seems to be a common pattern?  (At least I'm seeing the same for the Medication and Encounter translators).

Reply:
@mogoodrich In general, we expect the translation between versions to be handled by the provider... This should probably be something like:

```java
public class MedicationRequestFhirResourceProvider_2_2IntegrationTest extends BaseFhirR3IntegrationTest<MedicationRequestFhirResourceProvider, MedicationRequest> {
```

Where both `MedicationRequestFhirResourceProvider` and `MedicationRequest` are the R3 verisons of those classes. I might use the [`MedicationDispense` integration test as a model](https://github.com/openmrs/openmrs-module-fhir2/blob/master/integration-tests-2.6/src/test/java/org/openmrs/module/fhir2/provider/r3/MedicationDispenseFhirResourceProviderIntegrationTest.java).